lib/3scale/backend/storage_helpers.rb

```diff
@@ -98,9 +98,9 @@ def config_with(config,
               h[k] = val if val
             end.merge(options)
 
-            cfg_with_sentinels = cfg_sentinels_handler cfg
-
-            defaults.merge(ensure_url_param(cfg_with_sentinels))
+            cfg_compacted = cfg_compact cfg
+            cfg_with_sentinels = cfg_sentinels_handler cfg_compacted
+            cfg_defaults_handler cfg_with_sentinels, defaults
```

Comment:
As far as I could track this down, it appears to me correct.

It's not very clear for the reader though without thoroughly tracking what's going on.

It first handles sentinel options that don't work with the `url`, `path`, etc. options. And then we normalize these options regardless of config being sentinel or not.

I think it would be much better to have something like:
```suggestion
            cfg_compacted = cfg_compact cfg
            cfg_sentinels_handler(cfg_compacted) || cfg_defaults_handler(cfg_compact, defaults)
```

i.e. if sentinels config is found, then return it, otherwise handle as a non-sentinel config. Do not apply both handler one after the other. To make clear that these are very different things and avoid the possibility of introducing conflicts between the two handlers in the future.

Reply:
The call to `cfg_defaults_handler` is required in all scenarios. First, because it ensures some defaults are always sent to the client:

https://github.com/jlledom/apisonator/blob/054d0053e98e0a3fa82a99079867e7eba5446a35/lib/3scale/backend/storage_helpers.rb#L42-L59

It also ensures there's an `url` parameter, which is used from both sentinels and non-sentinels clients. For non-sentinels, it contains the redis URL; for sentinels, it contains the master name.

The only thing `cfg_defaults_handler` does which is exclusive for non-sentinels is translating `url: 'unix:///path/to/socket'` to `path: '/path/to/socket'`.